core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java

```diff
@@ -4310,6 +4310,16 @@ protected RelDataType validateSelectList(
     return typeFactory.createStructType(fieldList);
   }
 
+  private boolean containsDynamicStar(
+      List<Map.Entry<String, RelDataType>> fieldList) {
+    for (Map.Entry<String, RelDataType> item : fieldList) {
+      if (item.getValue().getSqlTypeName().getName().equals("DYNAMIC_STAR")) {
```

Comment:
Can we just compare getSqlTypeName() to the enum value of DYNAMIC_STAR rather than a string comparison?